### PR TITLE
1.0.0-rc.1 proposal

### PR DESCRIPTION
**Pull request by dyladan**
### :boom: Breaking Change

* [#55](https://github.com/open-telemetry/opentelemetry-js-api/pull/55) chore: move baggage methods in propagation namespace ([@vmarchaud](https://github.com/vmarchaud))
* [#65](https://github.com/open-telemetry/opentelemetry-js-api/pull/65) chore: remove suppress instrumentation ([@dyladan](https://github.com/dyladan))
* [#60](https://github.com/open-telemetry/opentelemetry-js-api/pull/60) chore: removing timed event ([@obecny](https://github.com/obecny))
* [#58](https://github.com/open-telemetry/opentelemetry-js-api/pull/58) chore: use spancontext for link ([@dyladan](https://github.com/dyladan))
* [#47](https://github.com/open-telemetry/opentelemetry-js-api/pull/47) chore: move span method for context in trace API #40 ([@vmarchaud](https://github.com/vmarchaud))
* [#45](https://github.com/open-telemetry/opentelemetry-js-api/pull/45) chore: rename `span#context()` to `span#spanContext` ([@dyladan](https://github.com/dyladan))
* [#43](https://github.com/open-telemetry/opentelemetry-js-api/pull/43) chore: renaming noop span to non recording span ([@obecny](https://github.com/obecny))
* [#32](https://github.com/open-telemetry/opentelemetry-js-api/pull/32) feat!: return boolean success value from setGlobalXXX methods ([@dyladan](https://github.com/dyladan))

### :rocket: Enhancement

* [#54](https://github.com/open-telemetry/opentelemetry-js-api/pull/54) feat: add tracer.startActiveSpan() ([@naseemkullah](https://github.com/naseemkullah))
* [#58](https://github.com/open-telemetry/opentelemetry-js-api/pull/58) chore: use spancontext for link ([@dyladan](https://github.com/dyladan))
* [#51](https://github.com/open-telemetry/opentelemetry-js-api/pull/51) feat: add function to wrap SpanContext in NonRecordingSpan #49 ([@dyladan](https://github.com/dyladan))

### :memo: Documentation

* [#64](https://github.com/open-telemetry/opentelemetry-js-api/pull/64) chore: document the reason for symbol.for ([@dyladan](https://github.com/dyladan))
* [#44](https://github.com/open-telemetry/opentelemetry-js-api/pull/44) chore: updating readme headline and fixing links ([@obecny](https://github.com/obecny))

### Committers: 6

* Bartlomiej Obecny ([@obecny](https://github.com/obecny))
* Daniel Dyla ([@dyladan](https://github.com/dyladan))
* Gerhard Stöbich ([@Flarna](https://github.com/Flarna))
* Naseem ([@naseemkullah](https://github.com/naseemkullah))
* Valentin Marchaud ([@vmarchaud](https://github.com/vmarchaud))
* t2t2 ([@t2t2](https://github.com/t2t2))

### CODE REVIEW

**Comment by Flarna** [comment]
I think we should either close or merge https://github.com/open-telemetry/opentelemetry-js-api/pull/46 first

**Comment by obecny** [comment]
this is missing the review and I think should be a part of this release -> https://github.com/open-telemetry/opentelemetry-js-api/pull/62

**Comment by dyladan** [comment]
@Flarna wants #46 to be merged first. @obecny wants core changes to be made in core before #46 can merge. Those core changes depend on unreleased API changes (we need `wrapSpan` to create noop spans without the noop tracer.

I fear we are in a dependency loop here. @Flarna @obecny how do you think we should resolve this ordering issue? My plan was to release rc.1 now, make the core changes, release core, then release rc.2 with noops removed.

**Comment by obecny** [comment]
> @Flarna wants #46 to be merged first. @obecny wants core changes to be made in core before #46 can merge. Those core changes depend on unreleased API changes (we need `wrapSpan` to create noop spans without the noop tracer.
> 
> I fear we are in a dependency loop here. @Flarna @obecny how do you think we should resolve this ordering issue? My plan was to release rc.1 now, make the core changes, release core, then release rc.2 with noops removed.

If we merge #46 we will have to do some work on core and contrib before people can use it and before we can release core and contrib. What I propose is to wait with merging #46 until it is done first in core and contrib so we don't block ourselves and that we have api that cannot be used with anything. This way we can do release on core and contrib quicker, and we are not blocked. We can do next rc as soon as all necessary changes are done in core and contrib. You merge #46 now and you will have api you cant use with anything for days.

**Comment by dyladan** [comment]
> > @Flarna wants #46 to be merged first. @obecny wants core changes to be made in core before #46 can merge. Those core changes depend on unreleased API changes (we need `wrapSpan` to create noop spans without the noop tracer.
> > I fear we are in a dependency loop here. @Flarna @obecny how do you think we should resolve this ordering issue? My plan was to release rc.1 now, make the core changes, release core, then release rc.2 with noops removed.
> 
> If we merge #46 we will have to do some work on core and contrib before people can use it and before we can release core and contrib. What I propose is to wait with merging #46 until it is done first in core and contrib so we don't block ourselves and that we have api that cannot be used with anything. This way we can do release on core and contrib quicker, and we are not blocked. We can do next rc as soon as all necessary changes are done in core and contrib. You merge #46 now and you will have api you cant use with anything for days.
> 
> unless I'm missing something so pls correct me :)

I'm fine with this, but I think there might already be some breaks that will need to be implemented in core anyways so it will be blocked no matter what. Holding off on #46 is fine as long as we can get these done quickly. If we merge this now, I'll create a PR to bump API in core immediately so we can make the changes required for #46

**Comment by dyladan** [comment]
any opinion here @Flarna?

**Comment by obecny** [comment]
> 
> I'm fine with this, but I think there might already be some breaks that will need to be implemented in core anyways so it will be blocked no matter what. Holding off on #46 is fine as long as we can get these done quickly. If we merge this now, I'll create a PR to bump API in core immediately so we can make the changes required for #46

If I understand correctly, after you do a new rc api release, we can do next release for core and contrib immediately, nothing needs to be changed except some renaming or using different namespace. All other changes has been done already right ?

**Comment by dyladan** [comment]
> > I'm fine with this, but I think there might already be some breaks that will need to be implemented in core anyways so it will be blocked no matter what. Holding off on #46 is fine as long as we can get these done quickly. If we merge this now, I'll create a PR to bump API in core immediately so we can make the changes required for #46
> 
> If I understand correctly, after you do a new rc api release, we can do next release for core and contrib immediately, nothing needs to be changed except some renaming or using different namespace. All other changes has been done already right ?

I think most but I'm not 100% sure.

**Comment by naseemkullah** [comment]
Will #54 be included as well since it has been merged?

**Comment by dyladan** [comment]
> Will #54 be included as well since it has been merged?

yes

**Comment by dyladan** [comment]
reviews pls :)

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,37 @@
...
+* [#65](https://github.com/open-telemetry/opentelemetry-js-api/pull/65) chore: remove suppress instrumentation ([@dyladan](https://github.com/dyladan))
+* [#60](https://github.com/open-telemetry/opentelemetry-js-api/pull/60) chore: removing timed event ([@obecny](https://github.com/obecny))
+* [#58](https://github.com/open-telemetry/opentelemetry-js-api/pull/58) chore: use spancontext for link ([@dyladan](https://github.com/dyladan))
+* [#47](https://github.com/open-telemetry/opentelemetry-js-api/pull/47) chore: move span method for context in trace API #40 ([@vmarchaud](https://github.com/vmarchaud))
+* [#45](https://github.com/open-telemetry/opentelemetry-js-api/pull/45) chore: rename `span#context()` to `span#spanContext` ([@dyladan](https://github.com/dyladan))
+* [#43](https://github.com/open-telemetry/opentelemetry-js-api/pull/43) chore: renaming noop span to non recording span ([@obecny](https://github.com/obecny))
+* [#32](https://github.com/open-telemetry/opentelemetry-js-api/pull/32) feat!: return boolean success value from setGlobalXXX methods ([@dyladan](https://github.com/dyladan))
+
+### :rocket: Enhancement
+
```
Comment (naseemkullah) [inline CHANGELOG.md:19]: #54 missing here

**Review by naseemkullah — CHANGES_REQUESTED** [review]

**Review by naseemkullah — APPROVED** [review]
apart from #54 missing in changelog lgtm

File: CHANGELOG.md (hunk 1 above)
Comment (dyladan) [reply CHANGELOG.md:19]: oops forgot to push

**Review by vmarchaud — APPROVED** [review]

**Comment by Flarna** [comment]
my intention was to reduce the number of breaking releases as they are usually not that nice for consumers.
But if rc.1 + core and contrib now and then more or less immediatelly rc.2 + core and contrib is the way to go it's fine for me.

**Review by Flarna — APPROVED** [review]
I think the readme should be updated (https://github.com/open-telemetry/opentelemetry-js-api/blob/main/README.md#100-rc0-to-x).

And I assume also the version in package.json needs an update. Or is this done automatically?

**Comment by dyladan** [comment]
OK. Waiting for @obecny because I think all maintainers should agree

**Comment by obecny** [comment]
> OK. Waiting for @obecny because I think all maintainers should agree

what about this one ?
https://github.com/open-telemetry/opentelemetry-js-api/pull/70#issuecomment-842503959

**Review by obecny — APPROVED** [review]
lgtm

**Comment by dyladan** [comment]
Tests and lint are passing locally. Is it ok if we merge and release this without the successful actions?

**Pull request merged**